### ISSUE TITLE
Logit caching includes padded tokens

### ISSUE THREAD

**Issue by FY-Liu** [r-three/SLMensembles#31]
When generating teacher logits, the output covers the maximum sequence length, resulting in redundant/random values beyond the actual input/label length for shorter sequences. These logits should be excluded.

Additionally, it is inefficient to save the entire padded dataset. The dataset should be unpadded prior to saving to reduce storage overhead and improve data efficiency.
